result/result.py

```diff
@@ -1,211 +1,245 @@
     """
-    def __init__(self, is_ok: bool, value: Union[T, E], force: bool = False) -> None:
-        """Do not call this constructor, use the Ok or Err class methods instead.
 
-        There are no type guarantees on the value if this is called directly.
+    @overload
+    def __init__(self) -> None:
+        pass
+
+    @overload
+    def __init__(self, value: T) -> None:
+        self._value = value
 
-        Args:
-            is_ok:
-                If this represents an ok result
-            value:
-                The value inside the result
-            force:
-                Force creation of the object. This is false by default to prevent
-                accidentally creating instance of a Result in an unsafe way.
+    def __init__(self, value: Any = True) -> None:
+        self._value = value
 
-        """
-        if force is not True:
-            raise RuntimeError("Don't instantiate a Result directly. "
-                               "Use the Ok(value) and Err(error) class methods instead.")
-        else:
-            self._is_ok = is_ok
-            self._value = value
+    def __repr__(self) -> str:
+        return "Ok({})".format(repr(self._value))
 
     def __eq__(self, other: Any) -> bool:
-        return (self.__class__ == other.__class__ and
-                self.is_ok() == cast(Result, other).is_ok() and
-                self._value == other._value)
+        return isinstance(other, Ok) and self.value == other.value
 
     def __ne__(self, other: Any) -> bool:
         return not (self == other)
 
     def __hash__(self) -> int:
-        return hash((self.is_ok(), self._value))
+        return hash((True, self._value))
 
-    def __repr__(self) -> str:
-        if self.is_ok():
-            return 'Ok({})'.format(repr(self._value))
-        else:
-            return 'Err({})'.format(repr(self._value))
+    def is_ok(self) -> bool:
+        return True
 
-    @classmethod
-    @overload
-    def Ok(cls) -> 'Result[bool, Any]':
-        pass
+    def is_err(self) -> bool:
+        return False
 
-    @classmethod
-    @overload
-    def Ok(cls, value: T) -> 'Result[T, Any]':
-        pass
+    def ok(self) -> T:
+        """
+        Return the value.
+        """
+        return self._value
+
+    def err(self) -> None:
+        """
+        Return `None`.
+        """
+        return None
 
-    @classmethod
-    def Ok(cls, value: Any = True) -> 'Result[Any, Any]':
-        return cls(is_ok=True, value=value, force=True)
+    @property
+    def value(self) -> T:
+        """
+        Return the inner value.
+        """
+        return self._value
+
+    def expect(self, _message: str) -> T:
+        """
+        Return the value.
+        """
+        return self._value
 
-    @classmethod
-    def Err(cls, error: E) -> 'Result[Any, E]':
-        return cls(is_ok=False, value=error, force=True)
+    def expect_err(self, _message: str) -> T:
```

Comment:
This return type could probably be `NoReturn` right?

(Nice find btw, I didn't know `NoReturn` previously :slightly_smiling_face:)

Reply:
Yup! Fixed in 1d52f3bb3ca6cc3faa37c6119fea2889cef4f1f7